src/JustEat.StatsD/Transport/DataExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace JustEat.StatsD
+{
+    public static class DataExtensions
+    {
+        public static ReadOnlySpan<byte> GetSpan(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span;
+            }
+            return src.Array.AsSpan();
+        }
+
+        public static byte[] GetArray(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span.ToArray();
```

Comment:
Would `Array.Empty<byte>()` be better as that just re-uses an already allocated array?

Reply:
Not sure I understood your point, so if Data block was created out of span, and consumer asked for an array, return an empty array?

This code will function correctly but will allocate on the heap. If we want to prevent allocation we could throw an exception.

Reply:
Ah, now I've re-read it, I see what's being done now.

Again though, this adds to my general hesitance around adding spans at this time that it seems to be over-complicating things.